pkg/appgateserver/cmd/cmd.go

```diff
@@ -0,0 +1,147 @@
+	cmd.Flags().StringVar(&flagListeningEndpoint, "listening-endpoint", "http://localhost:42069", "The host and port that the appgate server will listen on")
+	cmd.Flags().StringVar(&flagCometWebsocketUrl, "comet-websocket-url", "ws://localhost:36657/websocket", "The URL of the comet websocket endpoint to communicate with the pocket blockchain")
+	cmd.Flags().BoolVar(&flagSelfSigning, "self-signing", false, "Whether the server should sign all incoming requests with its own ring (for applications)")
+
+	cmd.Flags().String(flags.FlagKeyringBackend, "", "Select keyring's backend (os|file|kwallet|pass|test)")
+	cmd.Flags().String(flags.FlagNode, "tcp://localhost:36657", "The URL of the comet tcp endpoint to communicate with the pocket blockchain")
+
+	return cmd
+}
+
+func runAppGateServer(cmd *cobra.Command, _ []string) error {
+	// Create a context that is canceled when the command is interrupted
+	ctx, cancelCtx := context.WithCancel(cmd.Context())
+	defer cancelCtx()
+
+	// Handle interrupts in a goroutine.
+	go func() {
+		sigCh := make(chan os.Signal, 1)
+		signal.Notify(sigCh, os.Interrupt)
+
+		// Block until we receive an interrupt or kill signal (OS-agnostic)
+		<-sigCh
+		log.Println("INFO: Interrupt signal received, shutting down...")
+
+		// Signal goroutines to stop
+		cancelCtx()
+	}()
+
+	// Parse the listening endpoint.
+	listeningUrl, err := url.Parse(flagListeningEndpoint)
+	if err != nil {
+		return fmt.Errorf("failed to parse listening endpoint: %w", err)
+	}
+
+	// Setup the AppGate server dependencies.
+	appGateServerDeps, err := setupAppGateServerDependencies(cmd, ctx, flagCometWebsocketUrl)
+	if err != nil {
+		return fmt.Errorf("failed to setup AppGate server dependencies: %w", err)
+	}
+
+	log.Println("INFO: Creating AppGate server...")
+
+	// Create the AppGate server.
+	appGateServer, err := appgateserver.NewAppGateServer(
+		appGateServerDeps,
+		appgateserver.WithSigningInformation(&appgateserver.SigningInformation{
+			// provide the name of the key to use for signing all incoming requests
+			SigningKeyName: flagSigningKey,
+			// provide whether the appgate server should sign all incoming requests
+			// with its own ring (for applications) or not (for gateways)
+			SelfSigning: flagSelfSigning,
+		}),
+		appgateserver.WithListeningUrl(listeningUrl),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create AppGate server: %w", err)
+	}
+
+	log.Printf("INFO: Starting AppGate server, listening on %s...", listeningUrl.String())
+
+	// Start the AppGate server.
+	if err := appGateServer.Start(ctx); err != nil && !errors.Is(err, http.ErrServerClosed) {
+		return fmt.Errorf("failed to start app gate server: %w", err)
+	} else if errors.Is(err, http.ErrServerClosed) {
+		log.Println("INFO: AppGate server stopped")
+	}
+
+	return nil
+}
+
+func setupAppGateServerDependencies(cmd *cobra.Command, ctx context.Context, cometWebsocketUrl string) (depinject.Config, error) {
```

Comment:
@bryanchriswhite @red-0ne is this what you guys think is better?

Reply:
Yes, thank you.